### PR TITLE
Make the TriggeredByImage field ObjectReference

### PR DESCRIPTION
**Pull request by mfojtik**
@ncdc @liggitt @smarterclayton 

(also fixed overseen typo in the JSON serialization)

### CODE REVIEW

File: pkg/build/api/types.go [hunk 1]
```diff
@@ -407,8 +407,8 @@ type BuildRequest struct {
 	// Revision is the information from the source for a specific repo snapshot.
 	Revision *SourceRevision `json:"revision,omitempty"`
 
-	// TriggeredByImage is the image name that triggered this build.
-	TriggeredByImage string
+	// TriggeredByImage is the Image that triggered this build.
+	TriggeredByImage *kapi.ObjectReference `json:"triggeredByImage,omitempty"`
```
Comment (ncdc) [inline pkg/build/api/types.go]: No json on internal

File: pkg/build/controller/image_change_controller.go [hunk 2]
```diff
@@ -110,7 +110,10 @@ func (c *ImageChangeController) HandleImageRepo(repo *imageapi.ImageStream) erro
 				ObjectMeta: kapi.ObjectMeta{
 					Name: config.Name,
 				},
-				TriggeredByImage: triggeredImage,
+				TriggeredByImage: &kapi.ObjectReference{
+					Kind: "DockerImageReference",
```
Comment (ncdc) [inline pkg/build/controller/image_change_controller.go]: DockerImage

**Comment by mfojtik** [comment]
@ncdc fixed.

**Comment by ncdc** [comment]
LGTM. @smarterclayton PTAL

**Comment by smarterclayton** [comment]
LGTM [merge]

**Comment by mfojtik** [comment]
[merge]

**Pull request merged**